auth/config/config.go

```diff
@@ -0,0 +1,212 @@
+	AccessTokenLifespan                   config.Duration `json:"accessTokenLifespan" pflag:",Defines the lifespan of issued access tokens."`
+	RefreshTokenLifespan                  config.Duration `json:"refreshTokenLifespan" pflag:",Defines the lifespan of issued access tokens."`
+	AuthorizationCodeLifespan             config.Duration `json:"authorizationCodeLifespan" pflag:",Defines the lifespan of issued access tokens."`
+	ClaimSymmetricEncryptionKeySecretName string          `json:"claimSymmetricEncryptionKeySecretName" pflag:",OPTIONAL"`
+	TokenSigningRSAKeySecretName          string          `json:"tokenSigningRSAKeySecretName" pflag:",OPTIONAL: Secret name to use to retrieve RSA Signing Key."`
+	OldTokenSigningRSAKeySecretName       string          `json:"oldTokenSigningRSAKeySecretName" pflag:",OPTIONAL: Secret name to use to retrieve Old RSA Signing Key. This can be useful during key rotation to continue to accept older tokens."`
+
+	StaticClients map[string]*fosite.DefaultClient `json:"staticClients" pflag:"-,Defines statically defined list of clients to allow."`
+}
+
+type ExternalAuthorizationServer struct {
+	BaseURL config.URL `json:"baseUrl" pflag:",This should be the base url of the authorization server that you are trying to hit. With Okta for instance, it will look something like https://company.okta.com/oauth2/abcdef123456789/"`
+}
+
+type OAuth2Options struct {
+	AuthServerType     AuthorizationServerType     `json:"authServerType" pflag:"-,Determines authorization server type to use. Additional config should be provided for the chosen AuthorizationServer"`
+	SelfAuthServer     AuthorizationServer         `json:"selfAuthServer" pflag:",Authorization Server config to run as a service. Use this when using an IdP that does not offer a custom OAuth2 Authorization Server."`
+	ExternalAuthServer ExternalAuthorizationServer `json:"externalAuthServer" pflag:",External Authorization Server config."`
+	ThirdParty         ThirdPartyConfigOptions     `json:"thirdPartyConfig" pflag:",Defines settings to instruct flyte cli tools (and optionally others) on what config to use to setup their client."`
+}
+
+type UserAuthConfig struct {
+	// This is where the user will be redirected to at the end of the flow, but you should not use it. Instead,
+	// the initial /login handler should be called with a redirect_url parameter, which will get saved to a cookie.
+	// This setting will only be used when that cookie is missing.
+	// See the login handler code for more comments.
+	RedirectURL config.URL `json:"redirectUrl"`
+
+	OpenID OpenIDOptions `json:"openId" pflag:",OpenID Configuration for User Auth"`
+	// Possibly add basicAuth & SAML/p support.
+
+	CookieHashKeySecretName  string `json:"cookie_hash_key_secret_name" pflag:",OPTIONAL: Secret name to use for cookie hash key."`
```

Comment:
```suggestion
	CookieHashKeySecretName  string `json:"cookieHashKeySecretName" pflag:",OPTIONAL: Secret name to use for cookie hash key."`
```

Reply:
applied but didn't use the commit so I do not upset DCO